### DISCUSSION TITLE
BingAI on GCP Cloud Function

### DISCUSSION THREAD

**Discussion by VittorioOLG** [waylaidwanderer/node-chatgpt-api#276] [Help] [answered]
Hi there,

I'm trying to **run the BingAI module on a Firebase Function**, but I'm facing an issue. I've already extracted the BingAI code to make it as lean as possible, and **it works on Firebase emulators**. However, **when I deploy the code to GCP, I get an error**. Specifically, I'm **receiving an empty response body** when I make a request using **fetch**.
Thats the code, in _BingAIClient.js_, that is giving the error:

_"        const response = await fetch(`${this.options.host}/turing/conversation/create`, fetchOptions);     "_

Do you think the issue is because Google servers are being recognized by Bing? Would using a proxy help to fix this problem?

Thanks for your help!

**Comment by waylaidwanderer** [comment, marked as answer]
Yeah, that can be a sign of the IP being blocked. Try setting a proxy.

**Comment by VittorioOLG** [comment]
Hi Joel,
Thank you for the response!
It worked!!

To make it work, I used the WebShare proxy provider, which provides 10 proxies with a 1 GB bandwidth limit for free and requires no credit card.

To enable password-protected proxies, I made changes to the "BingAIClient.js" code as follows:

Since WebShare uses HTTP proxy, I changed HttpsProxyAgent to HttpProxyAgent:
const HttpProxyAgent = require('https-proxy-agent');

I updated the ProxyAgent inputs to allow password-protected proxies:
`
fetchOptions.dispatcher = new ProxyAgent({ 
     uri: 'http://<PROXY-URI>', //Proxy example.: 123.123.123.123:1234 
     secureProxy: true, 
     headers: { 'Proxy-Authorization': 'Basic ' + Buffer.from('<USERNAME>:<PASSWORD>').toString('base64') } });
`

I changed HttpsProxyAgent to HttpProxyAgent with some input changes:
`
new HttpProxyAgent('http://<USERNAME>:<PASSWORD>@<PROXY>'); //Proxy example: 123.123.123.123:1234
`

I hope this information is useful to others who may be encountering similar challenges.
